### ISSUE TITLE
Tethering doesn't seem to go through orbot

### ISSUE THREAD

**Issue by cjeanneret** [EthACKdotOrg/orWall#32]
@hxe reported, as a side comment, tethering doesn't seem to go through orbot with newer versions (see #29).
There are apparently multiple problems:
- [ ] tethering doesn't go through orbot
- [ ] tethering seems  seems to make all the apps bypass orbot (probably a problem in _nat_ table)
- [ ] when tethering is disabled in orwall, it seems to break the whole rule-set

**Comment by eadf5146** [comment]
Hi, has there been any progress on this, or is there an update? I recently have been trying to tether through orbot without success.

I need to tether to my phone and run the whole internet connection through orbot. Any device which tethers to my phone MUST have all it's traffic go through the Tor network. I have tried multiple different options and different versions of orbot but no matter what I try so far, tethered traffic will not go through TOR and the ip is of the non torified traffic. Usually other apps on the phone
do go through the TOR network though, so that part works at least. I have very limited time available to try other endless possibilities so could someone please clarify what is known to work?

If the newest version of orbot still does not work for running tethering through orbot, which is the best older version to downgrade to to get it to work?

Also, do I perhaps need to use Orwall as well to get it to work? I have not had a lot of time to spend on trying different possible combinations with that yet. 
How should I set the options in Orwall? If I need to set things in a certain order to get around any bugs, what is this? Should I use my phones built in tethering system to set up the wifi hotspot, or a custom app? I tried it with the built in tethering feature, and also barnacle without success. Could the phone make and model be an issue or the version of android be incompatible somehow? It is a sony experia C1505 running Android 4.1.1

The current version of orbot I have tried it with is 14.1.4 - I am also confused what versions of orbot people are referring to on here as people keep talking about versions starting 0.20 or whatever. I assume either you are putting an extra 0 in front of the version number is it usually
referred to as elsewhere? (so you would refer to 14.1.4 as 0.14.1.4), or are you talking about versions of orwall? I have version 1.0.6 of orwall. Is the version of this likely to be a possible factor? lol.

All very confusing, so any clarity would be appreciated.

Thanks

**Comment by cjeanneret** [comment]
Hello,

hmm, I think the main confusion is due to names and application usages:
orbot, mostly maintained by @n8fr8 from GuardianProject, is the client app for Tor. It has some capabilities for traffic redirect and so on. For this one, versions are 14.xx for stable, 15.xx for RC, and so on.

orwall, on the other hand, is "only" a firewall application, deeply linked to orbot. Versions were 0.x.y until "stable" release, which is now 1.x.y

That said: tethering part isn't the easiest thing, as there are system firewall rules being applied on the fly, meaning we first must understand fully what have been added, which network interfaces are affected by the rules, what do the rules do.
We already had a hard time allowing tethering without having to disable orwall (this part isn't really stable for now), and we will also need to dig a bit further in order to see, once a device is connected, how its traffic is redirected on the device.

Also, I had some discussions with some Tor Project people regarding the "good and bad" for some kind of "anonybox", which will be what your smartphone will become in your use case — it seems there are pros and cons, among cons "bad idea to redirect all your traffic through tor, especially if you don't know what app access network nor what they do on it". And I must agree with this statement — doesn't mean I won't try to make it work, just it might be an option you must enable.

You might prefer some tor entry on your computer/tethered device instead, and doing the proper configurations on it, at least for now.

TLDR;
Status for now: not working for this particular part. Tethering isn't easy to manage due to system iptables rules being added, new network device being created/activated and so on.

Cheers,

C.
